ui/src/auth/auth0OAuth.tsx

```diff
@@ -0,0 +1,163 @@
+
+  useEffect(() => {
+    if (shouldNavigate) {
+      console.info("Return to previous page: " + reason);
+      const from: Path | undefined = location.state?.from;
+      navigate(from || "/", { replace: true });
+    }
+  }, [location, navigate]);
+}
+
+export function LoginPage() {
+  const { error, isAuthenticated, isLoading, loginWithRedirect } = useAuth0();
+  const location = useLocation();
+
+  returnToPreviousPage(!isAuth0Enabled(), "Auth0 login not enabled");
+  returnToPreviousPage(isAuthenticated, "Already authenticated");
+
+  const signIn = useCallback(
+    async (from?: string) => {
+      await loginWithRedirect({ appState: { returnTo: from } });
+    },
+    [loginWithRedirect]
+  );
+
+  return (
+    <Box>
+      <GridLayout
+        cols
+        rowAlign="middle"
+        colAlign="center"
+        sx={{ px: 2, py: 1, minHeight: "200px" }}
+      >
+        <GridLayout
+          colAlign="center"
+          spacing={1}
+          height="auto"
+          sx={{ textAlign: "center" }}
+        >
+          <img src={verilyImage} style={{ width: "100px", height: "auto" }} />
+          <ErrorList errors={error} />
+          <Button
+            variant="contained"
+            onClick={() => signIn(location.state?.from)}
+            disabled={isLoading}
+          >
+            Sign in to Data Explorer
+          </Button>
+        </GridLayout>
+      </GridLayout>
+    </Box>
+  );
+}
+
+export const LogoutPage = () => {
+  returnToPreviousPage(!isAuth0Enabled(), "Auth0 logout not enabled");
+
+  const { error, isLoading, logout } = useAuth0();
+
+  const signOut = useCallback(() => {
+    logout({ logoutParams: { returnTo: window.location.origin } });
+  }, [logout]);
+
+  return (
+    <Box>
+      <GridLayout
+        cols
+        rowAlign="middle"
+        colAlign="center"
+        sx={{ px: 2, py: 1, minHeight: "200px" }}
+      >
+        <GridLayout
+          colAlign="center"
+          spacing={1}
+          height="auto"
+          sx={{ textAlign: "center" }}
+        >
+          <img src={verilyImage} style={{ width: "100px", height: "auto" }} />
+          <ErrorList errors={error} />
+          <Button
+            variant="contained"
+            onClick={() => signOut()}
+            disabled={isLoading}
+          >
+            Sign out of Data Explorer
+          </Button>
+        </GridLayout>
+      </GridLayout>
+    </Box>
+  );
+};
+
+export function getAuth0Token(
+  loginType: loginAccessType
+): () => Promise<string> {
+  const { isAuthenticated, getAccessTokenSilently } = useAuth0();
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  // Not authenticated, redirect to login
+  useEffect(() => {
+    if (!isAuthenticated) {
+      if (loginType == loginAccessType.RedirectUrl) {
+        redirect("/login");
+      } else if (loginType == loginAccessType.NavigatePath) {
+        navigate("login", { replace: true, state: location });
```

Comment:
done

Reply:
reverted it , since "/" prefix might imply that it is an absolute path. 
tested - it is a relative path always.

Reply:
That seems surprising since we're using that behavior in other places (e.g. `newCriteria.tsx:25`) and I see it referenced in places like [this question](https://stackoverflow.com/questions/69990531/react-router-v6-navigate-to-absolute-path-from-subrouter), though the linked doc doesn't seem to say it anymore. Maybe there's something else going on there.

Reply:
removed in latest changeset